cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -509,6 +517,8 @@ runTxBuild (AnyCardanoEra era) (AnyConsensusModeParams cModeParams) networkId mS
         newExceptT . fmap (join . first ShelleyTxCmdAcquireFailure) $
           executeLocalStateQueryExpr localNodeConnInfo Nothing $ \_ntcVersion -> runExceptT $ do
             unless (null txinsc) $ do
+              -- TODO: Question, why do we not need the collateralUtxo to be included in
+              -- the utxo?
```

Comment:
I'm confused by this as well, the ledger rules do require that the collateral utxo are legitimate.

Reply:
Yep I'm not sure why we don't have to include them in the `utxo`. But I will diagnose that some other time.